src/Template/Layout/js/app/components/filter-box.js

```diff
@@ -161,17 +162,66 @@ export default {
          *
          * @emits Event#filter-objects
          */
-        onQueryStringChange() {
-            let queryString = this.queryFilter.q || "";
+        onQueryStringKeyup() {
+            const queryString = this.queryFilter.q || "";
 
             clearTimeout(this.timer);
-            if (queryString.length >= 3 || queryString.length == 0) {
+            if (queryString.trim().length >= 3 || queryString.trim().length === 0) {
                 this.timer = setTimeout(() => {
                     this.$emit("filter-objects", this.queryFilter);
                 }, 300);
             }
         },
 
+        /**
+         * Handle Search input when value changes
+         * If search text is empty or at least 3 characters long, ok.
+         * Otherwise, warn that search text is too short.
+         */
+        onQueryStringChange() {
+            if (!this.searchFieldValid()) {
+                this.searchFieldDialog();
+            }
+        },
+
+        /**
+         * Verify search text.
+         * If is empty or more than 2 characters long, return true.
+         * Return false otherwise.
+         */
+        searchFieldValid() {
+            const search = document.getElementById('search');
+            if (!search) {
+                return true;
+            }
+            search.value = search.value.trim();
+            if (!this.queryFilter) {
+                this.queryFilter = {};
+            }
+            this.queryFilter.q = search.value;
+
+            return (search.value.length === 0 || search.value.length > 2);
+        },
+
+        /**
+         * Verify search text.
+         * If is empty or more than 2 characters long, ok.
+         * Show prompt dialog otherwise.
+         */
+        searchFieldDialog() {
+            const dialog = this.$root.$refs.beditaDialog;
+            dialog.cancelMessage = '';
+            const confirmCallback = (newVal) => {
+                search.value = newVal.trim();
+                if (!this.queryFilter) {
+                    this.queryFilter = {};
+                }
+                this.queryFilter.q = search.value;
+                dialog.hide();
+            };
+            dialog.prompt(t`Search text too short. Minimum length is 3. Retry`, search.value, confirmCallback);
```

Comment:
[minor] search value in dialog could be read-only or simply added to the message - otherwise you may think of change it and relaunch search from the dialog (value is editable...)

Reply:
you can change it with prompt, actually

Reply:
let's simplify. warning instead of prompt